cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,401 @@
+			if err != nil {
+				t.Errorf("Coudn't create the calculation: %v", newCalculation)
+			} else {
+				logrus.WithFields(logrus.Fields{"teff": teff, "logG": "4.0"}).Info("Calculation successfully created")
+			}
+			tc.expected = append(tc.expected, *newCalculation)
+		}
+
+		s := server{
+			logger: logrus.WithField("test-name", tc.id),
+			ctx:    ctx,
+			client: fakeClient,
+		}
+
+		req, err := http.NewRequest("GET", "/calculations", nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		req.Header.Set("Content-Type", "application/json")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(s.getCalculations)
+		handler.ServeHTTP(rr, req)
+		if status := rr.Code; status != http.StatusOK {
+			t.Errorf("Handler returned wrong status code: got %v want %v",
+				status, http.StatusOK)
+		}
+
+		var actualData *calculationsv1.CalculationList
+
+		err = json.Unmarshal(rr.Body.Bytes(), &actualData)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if diff := cmp.Diff(tc.expected, actualData.Items); diff != "" {
+			t.Fatal(diff)
+		}
+	}
+}
+
+func TestGetCalculationByName(t *testing.T) {
+	testCases := []struct {
+		id               string
+		calculationCount int
+		errorExpected    bool
+		expected         *calculationsv1.Calculation
+	}{
+		{
+			id:               "no calculation",
+			calculationCount: 0,
+			errorExpected:    true,
+		},
+		{
+			id:               "one calculation gets created",
+			calculationCount: 1,
+		},
+	}
+
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	teff := 10000
+	for _, tc := range testCases {
+		var newCalculation *calculationsv1.Calculation
+		var calculationCount = teff + tc.calculationCount
+		for teff != calculationCount {
+			teff++
+			newCalculation = util.NewCalculation(float64(teff), 4.0)
+			logrus.WithFields(logrus.Fields{"teff": teff, "logG": "4.0"}).Info("Creating calculation for test purpose")
+			_, err := fakeClient.Calculations().Create(ctx, newCalculation, metav1.CreateOptions{})
+			if err != nil {
+				t.Errorf("Coudn't create the calculation: %v", newCalculation)
+			} else {
+				logrus.WithFields(logrus.Fields{"name": newCalculation.Name, "teff": teff, "logG": "4.0"}).Info("Calculation successfully created")
+			}
+			tc.expected = newCalculation
+		}
+
+		s := server{
+			logger: logrus.WithField("test-name", tc.id),
+			ctx:    ctx,
+			client: fakeClient,
+		}
+
+		req, err := http.NewRequest("GET", fmt.Sprintf("/calculation/%v", newCalculation.Name), nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if tc.calculationCount > 0 {
+			q := req.URL.Query()
+			q.Add("name", newCalculation.Name)
+			req.URL.RawQuery = q.Encode()
+		}
+
+		req.Header.Set("Content-Type", "application/json")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(s.getCalculationByName)
+		handler.ServeHTTP(rr, req)
+
+		response := Response{}
```

Comment:
This is wrong. The `getCalculationByName()` returns an actual calculation. In case if you need to check for error, just check the StatusCode.

Reply:
Ah I see that you already doing that below. I guess you are trying here to capture the status code from the JSON output. You should get the status code from the headers, but you have to make the function to update the headers.